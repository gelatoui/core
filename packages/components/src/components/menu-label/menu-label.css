.label-container {
  display: flex;
  padding: var(--spacing-100, 0.5rem) var(--spacing-200, 1rem);
  justify-content: center;
  align-items: center;
  color: var(--tokens-text-on-light-default, #4b515a);

  font-family: var(--body-medium-regular-font-family, 'Inter');
  font-size: var(--body-medium-regular-font-size, 1rem);
  font-style: normal;
  font-weight: 400;
  line-height: var(--body-medium-regular-line-height, 1.25rem);
  letter-spacing: var(--body-medium-regular-letter-spacing, 0);
  text-decoration: none;
  transition: color 0.2s ease;
}

a.label-container:hover {
  color: var(--tokens-text-on-light-hover, #0a5bf8);
}
