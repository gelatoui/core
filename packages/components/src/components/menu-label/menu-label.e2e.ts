/* eslint-disable no-undef */
import { newE2EPage } from '@stencil/core/testing'

describe('glu-menu-label e2e', () => {
  it('renders an anchor when href is provided', async () => {
    const page = await newE2EPage()

    await page.setContent('<glu-menu-label href="https://example.com">Link Label</glu-menu-label>')

    const element = await page.find('glu-menu-label')

    expect(element).not.toBeNull()

    // Using the shadow piercing selector (>>>)
    const anchor = await page.find('glu-menu-label >>> a')

    expect(anchor).not.toBeNull()

    expect(anchor.getAttribute('href')).toBe('https://example.com')
  })

  it('renders slot content when no href is provided', async () => {
    const page = await newE2EPage()

    await page.setContent('<glu-menu-label>Plain Label</glu-menu-label>')

    const element = await page.find('glu-menu-label')

    expect(element).not.toBeNull()

    // Verify that the default slot content is present in the shadow root
    const slot = await page.find('glu-menu-label >>> slot')

    expect(slot).not.toBeNull()
  })
})
