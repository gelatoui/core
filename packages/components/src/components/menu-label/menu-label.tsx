import { Component, h, Host, Prop } from '@stencil/core'

/**
 * @component
 * @tag glu-menu-label
 * @shadow true
 *
 * @slot - The text content of the menu label.
 */
@Component({
  tag: 'glu-menu-label',
  styleUrl: 'menu-label.css',
  shadow: true
})
export class GluMenuLabel {
  /**
   * URL to navigate to when clicking the label.
   */
  @Prop() readonly href?: string

  /**
   * Specifies where to open the linked document. Only applicable if `href` is set.
   */
  @Prop() readonly target?: '_self' | '_blank' | '_parent' | '_top'

  /**
   * Specifies the relationship between the current document and the linked document. Only applicable if `href` is set.
   */
  @Prop() readonly rel?: string

  render() {
    return (
      <Host class="glu-menu-label" part="menu-label">
        {this.href ?
          (
            <a href={this.href} target={this.target} rel={this.rel} class="label-container">
              <slot></slot>
            </a>
          ) :
          (
            <span class="label-container">
              <slot></slot>
            </span>
          )}
      </Host>
    )
  }
}
