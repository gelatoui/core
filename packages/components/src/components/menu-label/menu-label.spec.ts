/* eslint-disable no-undef */
import { GluMenuLabel } from './menu-label'

import { newSpecPage } from '@stencil/core/testing'

describe('glu-menu-label', () => {
  it('renders slot content when no href is provided', async () => {
    const page = await newSpecPage({
      components: [GluMenuLabel],
      html: '<glu-menu-label>Test Label</glu-menu-label>'
    })

    // In shadow DOM, the default slot is rendered automatically.
    expect(page.root).toEqualHtml(`
      <glu-menu-label class="glu-menu-label">
        <mock:shadow-root>
          <slot></slot>
        </mock:shadow-root>
        Test Label
      </glu-menu-label>
    `)
  })

  it('renders an anchor tag when href is provided', async () => {
    const page = await newSpecPage({
      components: [GluMenuLabel],
      html: '<glu-menu-label href="https://example.com">Link Label</glu-menu-label>'
    })

    const anchor = page.root.shadowRoot.querySelector('a')

    expect(anchor).not.toBeNull()

    expect(anchor.getAttribute('href')).toBe('https://example.com')
  })

  it('forwards inherited attributes to the inner anchor element', async () => {
    const page = await newSpecPage({
      components: [GluMenuLabel],
      html: '<glu-menu-label href="https://example.com" title="Test Title">Link Label</glu-menu-label>'
    })

    const anchor = page.root.shadowRoot.querySelector('a')

    expect(anchor).not.toBeNull()

    // Verify that the title attribute is forwarded (assuming inheritAttributes works as expected)
    expect(anchor.getAttribute('title')).toBe('Test Title')
  })
})
