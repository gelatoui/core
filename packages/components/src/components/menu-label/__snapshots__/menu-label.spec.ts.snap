// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`glu-menu-item renders correctly as a button 1`] = `
<glu-menu-item
  type="button"
>
  Click Me
</glu-menu-item>
`;

exports[`glu-menu-item renders correctly as a label 1`] = `
<glu-menu-item
  type="label"
>
  Label Item
</glu-menu-item>
`;
